### PR TITLE
Fix:  build with `QTSERIALPORT=false`

### LINKED ISSUE #56944: Compile error WITH_SERIALPORT=false
### What is the bug or the crash?

Compile error `WITH_SERIALPORT=false` 
```
sip-build: /var/tmp/portage/sci-geosciences/qgis-3.36.1/work/qgis-3.36.1_build/python/core/core.sip: line 101: column 9: 'QtSerialPort/QtSerialPortmod.sip' could not be found
ninja: build stopped: subcommand failed.
```

### Steps to reproduce the issue

compile 3.36.1 using option `WITH_SERIALPORT=false`
- configure 
  ```
  cmake -C /var/tmp/portage/sci-geosciences/qgis-3.36.1/work/qgis-3.36.1_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DQGIS_MANUAL_SUBDIR=share/man/ -DQGIS_LIB_SUBDIR=lib64 -DQGIS_PLUGIN_SUBDIR=lib64/qgis -DQWT_INCLUDE_DIR=/usr/include/qwt6 -DQWT_LIBRARY=/usr/lib64/libqwt6-qt5.so -DQGIS_QML_SUBDIR=/usr/lib64/qt5/qml -DPEDANTIC=OFF -DUSE_CCACHE=OFF -DWITH_ANALYSIS=ON -DWITH_APIDOC=OFF -DWITH_GUI=ON -DWITH_INTERNAL_MDAL=ON -DWITH_QSPATIALITE=yes -DENABLE_TESTS=yes -DWITH_3D=yes -DWITH_GSL=yes -DCMAKE_DISABLE_FIND_PACKAGE_HDF5=OFF -DWITH_SERVER=yes -DCMAKE_DISABLE_FIND_PACKAGE_NetCDF=OFF -DUSE_OPENCL=yes -DWITH_ORACLE=no -DWITH_QWTPOLAR=no -DWITH_PDAL=yes -DWITH_COPC=yes -DWITH_POSTGRESQL=yes -DWITH_BINDINGS=yes -DWITH_CUSTOM_WIDGETS=yes -DWITH_QUICK=yes -DWITH_QTSERIALPORT=no -DWITH_QTWEBKIT=OFF -DBUILD_WITH_QT6=no -DGRASS_PREFIX8=/usr/lib64/grass83 -DWITH_GRASS8=ON -DBINDINGS_GLOBAL_INSTALL=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage/sci-geosciences/qgis-3.36.1/work/qgis-3.36.1_build/gentoo_toolchain.cmake /var/tmp/portage/sci-geosciences/qgis-3.36.1/work/qgis-3.36.1
  ```
- compile 
  ```
  ninja -v j33 -l0
  ```

### Versions

- 3.36.1
- or basically any version since https://github.com/qgis/QGIS/commit/9195dbd05750498f08a473bdd4ac23e660743531 fails with this very error

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

cc @jtornero this issue seems to be related to 9195dbd05750498f08a473bdd4ac23e660743531 which assumes unconditionally `WITH_SERIALPORT=true` - which isn't the case. AFAICT QtSerialport should be compiled conditionally only upon `%If (HAVE_QTSERIALPORT)` as re-fixed by https://github.com/qgis/QGIS/commit/ecdb1e2989e3bf86978590fc4fd724b18b8dde27. 
this concerns at least https://github.com/qgis/QGIS/commit/9195dbd05750498f08a473bdd4ac23e660743531#diff-9ad07175b4e5bd06fbdf8f4a8f69b63d4c674a3bbcca791650067f051c761be5R101 and https://github.com/qgis/QGIS/commit/9195dbd05750498f08a473bdd4ac23e660743531#diff-a71aa025b44462573371fd6c18befdff13db223ff9736f1dfea29fd78774902eR101 and probably also the python code

... or would you vote to drop support for this option and make qtserial a requirement?

### PR DESCRIPTION
**Pull request by t0b3**
## Description

- fixes build error for `QTSERIALPORT=false`
- disable uploading of docker image if `github.actor != 'qgis'` so GH actions in forked repos won't fail for this reason

Backport
- [ ] backport to 3.36 releases

#Fixes #56944

### CODE REVIEW

File: python/core/core.sip.in [hunk 1]
```diff
@@ -91,21 +91,24 @@ done:
 
 %End
 
+%Feature HAVE_GUI
+%Feature HAVE_QTSERIALPORT
```
Comment (m-kuhn) [inline python/core/core.sip.in:95]: Do we need all the other features?

**Comment by m-kuhn** [comment]
Thanks, for reference, what problem are you fixing with the github actor change?

File: python/core/core.sip.in (hunk 1 above)
Comment (t0b3) [reply python/core/core.sip.in:95]: @m-kuhn  hmm, that's a good question - It was already there... just tell me if you feel some of them should be removed in this PR as well :+1:

Comment (m-kuhn) [reply python/core/core.sip.in:95]: They were not there, that's why they are green (added) in https://github.com/qgis/QGIS/pull/56953/files

Can you keep this clean?

Comment (m-kuhn) [reply python/core/core.sip.in:95]: Oh I see, you moved them up

**Review by m-kuhn — APPROVED** [review]
Lgtm

File: python/core/core.sip.in (hunk 1 above)
Comment (t0b3) [reply python/core/core.sip.in:95]: @m-kuhn sure they were ... just a few lines lower ... and the need to be introduced before `%If (<your_feature>)` that's why I suggest to shift them all together

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -116,7 +116,7 @@ jobs:
           echo QT_VERSION: ${QT_VERSION}
 
       - name: Login to Docker Hub
-        if: ${{ github.event_name == 'push' }}
+        if: ${{ github.event_name == 'push' && github.actor == 'qgis' }}
```
Comment (3nids) [inline .github/workflows/run-tests.yml:119]: This caused that the image hasn't been uploaded since then

Comment (3nids) [reply .github/workflows/run-tests.yml:119]: fixed in #60200

Comment (t0b3) [reply .github/workflows/run-tests.yml:119]: @3nids thank you !

Comment (m-kuhn) [reply .github/workflows/run-tests.yml:119]: thanks @3nids 
should the same adjustment also be applied for all the other changes from this PR?

Comment (3nids) [reply .github/workflows/run-tests.yml:119]: thanks Matthias, done in 0ebaa3986914d6e588c5251d26fb7120edbabd30

**Pull request merged**